File: javascript/BAEKJOON/BRONZE Ⅰ/10798 세로읽기.js
var input = require('fs').readFileSync(__dirname+'/test.txt').toString().split('\n');
//var input = require('fs').readFileSync('/dev/stdin').toString().toString().split('\n');

let strings = []
strings.push([]);

// 5줄의 입력을 처리
for (let i = 0; i < 5; i++) {
    const line = input[i];  // i번째 줄 읽기

    // 각 줄의 문자를 세로로 저장
    for (let j = 0; j < line.length; j++) {
        // 필요한 경우 새로운 열(배열) 추가
        if(strings.length < (j + 1)){
            strings.push([]);
        }
        // j번째 열에 현재 문자 추가
        strings[j].push(line[j])
    }
}

let result = ""           // 결과 문자열을 저장할 변수 초기화

// 모든 열(세로로 읽은 문자들)을 순서대로 연결
for (let i = 0; i < strings.length; i++) {
    // 각 열의 문자들을 공백 없이 연결하여 결과 문자열에 추가
    result += strings[i].join("");
}
console.log(result);

File: Python/BAEKJOON/BRONZE Ⅰ/10798 세로읽기.py
strings = [[]]  # 2차원 리스트 초기화, 세로로 읽은 문자들을 저장할 리스트

# 5줄의 입력을 처리
for _ in range(5):
    line = input()  # 한 줄 읽기

    # 각 줄의 문자를 세로로 저장
    for i in range(len(line)):
        # 필요한 경우 새로운 열(리스트) 추가
        if len(strings) < (i + 1):
            strings.append([])
        # i번째 열에 현재 문자 추가
        strings[i].append(line[i])

# 결과 출력: 모든 열을 먼저 내부적으로 조인한 후, 외부적으로 조인하여 최종 문자열 생성
print(''.join([''.join(line) for line in strings]))